spark/src/main/scala/org/apache/spark/sql/delta/catalog/DeltaCatalog.scala

```diff
@@ -692,6 +699,22 @@ class DeltaCatalog extends DelegatingCatalogExtension
           })
         }
 
+        // Any ColumnChange not explicitly on the allowlist is blocked from making changes on
+        // Identity Columns
+        val disallowedColumnChangesOnIdentityColumns = columnChanges.filterNot {
+            case _: UpdateColumnComment | _: UpdateColumnPosition | _: RenameColumn
+                 | _: SyncIdentity => true
```

Comment:
Looks like we'll have to merge https://github.com/delta-io/delta/pull/3005 first.